test/libsolidity/syntaxTests/operators/custom/operator_not_user_implementable.sol

```diff
@@ -0,0 +1,23 @@
+using {
+    f as <<,
+    f as >>,
+    f as **,
+    f as ++,
+    f as !,
+    f as x,
+    f as delete,
+    f as new,
+    f as ()
+} for int256;
+
+// ----
+// ParserError 4403: (17-19): The operator << cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (30-32): The operator >> cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (43-45): The operator ** cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (56-58): The operator ++ cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (69-70): The operator ! cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
```

Comment:
Why is it not possible to redefine these? Wouldn't it be nice to be able to redefine those too?

Reply:
Actually, we can easily allow to redefine `<<`,` >>`, `!` and `**`. But, for ++ and -- an extra effort would be required. That's because these two are supposed to modify LValues they are called on.

Reply:
Code updated.

Reply:
I thought the issue with `<<`, `>>` and `**` was that their arguments and/or return types are not necessarily the same and they have some restrictions. For example the right argument cannot be signed. And `!` is a boolean operator so it needs to return `bool`.

Can we really allow them without breaking anything - or at least, do you have test coverage for their usage with those non-uniform types?